lntest/btcd.go

```diff
@@ -0,0 +1,67 @@
+package lntest
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/integration/rpctest"
+	"github.com/btcsuite/btcd/rpcclient"
+)
+
+// BtcdBackendConfig is an implementation of the BackendConfig interface
+// backed by a btcd node.
+type BtcdBackendConfig struct {
+	rpcConfig rpcclient.ConnConfig
+	harness   *rpctest.Harness
+}
+
+// GenArgs returns the arguments needed to be passed to LND at startup for
+// using this node as a chain backend.
+func (b BtcdBackendConfig) GenArgs() []string {
+	var args []string
+	encodedCert := hex.EncodeToString(b.rpcConfig.Certificates)
+	args = append(args, "--bitcoin.node=btcd")
+	args = append(args, fmt.Sprintf("--btcd.rpchost=%v", b.rpcConfig.Host))
+	args = append(args, fmt.Sprintf("--btcd.rpcuser=%v", b.rpcConfig.User))
+	args = append(args, fmt.Sprintf("--btcd.rpcpass=%v", b.rpcConfig.Pass))
+	args = append(args, fmt.Sprintf("--btcd.rawrpccert=%v", encodedCert))
+
+	return args
+}
+
+// P2PAddr returns the address of this node to be used when connection over the
+// Bitcoin P2P network.
+func (b BtcdBackendConfig) P2PAddr() string {
+	return b.harness.P2PAddress()
+}
+
+// NewBtcdBackend starts a new rpctest.Harness and returns a BtcdBackendConfig
+// for that node.
+func NewBtcdBackend() (*BtcdBackendConfig, func(), error) {
+	args := []string{
+		"--rejectnonstd",
+		"--txindex",
+		"--trickleinterval=100ms",
+	}
+	netParams := &chaincfg.RegressionNetParams
+	chainBackend, err := rpctest.New(netParams, nil, args)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to create btcd node: %v", err)
+	}
+
+	if err := chainBackend.SetUp(false, 0); err != nil {
```

Comment:
We no longer need to again call: https://godoc.org/github.com/btcsuite/btcd/rpcclient#Client.NotifyNewTransactions here? This is what allow us to be notified of things like inclusion of a transaction into the mempool. Ah, actually this isn't needed if we wait on the miner to have these transactions rather than the backend that `lnd` is connected to.

Reply:
Yes, this is called for the miner, which is enough for the test framework.